src/main/java/com/nitnelave/CreeperHeal/block/CreeperContainer.java

```diff
@@ -0,0 +1,140 @@
+        this.dependents.add(right);
+
+    }
+
+    /*
+     * @see com.nitnelave.CreeperHeal.block.Replaceable#remove()
+     */
+    @Override
+    public void remove()
+    {
+        World world = blockState.getWorld();
+        if (CreeperConfig.getWorld(world).getBool(WCfgVal.DROP_CHEST_CONTENTS))
+        {
+            Location location = blockState.getLocation().add(0.5, 0.5, 0.5);
+            for (ItemStack itemStack : ((Container) blockState).getSnapshotInventory().getContents())
+                if (itemStack != null)
+                    world.dropItemNaturally(location, itemStack);
+            ((Container) blockState).getSnapshotInventory().clear();
+            blockState.update(true, false);
+            for (BlockState dependent : dependents)
+            {
+                if (!(dependent instanceof Container))
+                    continue;
+                location = dependent.getLocation().add(0.5, 0.5, 0.5);
+                for (ItemStack itemStack : ((Container) dependent).getSnapshotInventory().getContents())
+                    if (itemStack != null)
+                        world.dropItemNaturally(location, itemStack);
+                ((Container) dependent).getSnapshotInventory().clear();
+                dependent.update(true, false);
+            }
+        }
+
+        ((Container) blockState).getInventory().clear();
+        for (BlockState dependent : dependents) 
+            if (dependent instanceof Container)
+                ((Container) dependent).getInventory().clear();
+
+        super.remove();
+    }
+
+    /*
+     * @see com.nitnelave.CreeperHeal.block.Replaceable#drop(boolean)
+     */
+    @Override
+    public boolean drop(boolean forced)
+    {
+        if (forced || CreeperConfig.shouldDrop())
+        {
+            BlockState current = blockState.getBlock().getState();
+            blockState.update(true, false);
+            Collection<ItemStack> drop = blockState.getBlock().getDrops();
+            current.update(true, false);
+            Location location = blockState.getLocation().add(0.5, 0.5, 0.5);
+            World world = blockState.getWorld();
+
+            for (ItemStack s : drop)
+                world.dropItemNaturally(location, s);
+
+            for (BlockState dependent : dependents)
+            {
+                current = dependent.getBlock().getState();
+                dependent.update(true, false);
+                drop = dependent.getBlock().getDrops();
+                current.update(true, false);
+                location = dependent.getLocation().add(0.5, 0.5, 0.5);
+                world = dependent.getWorld();
+                for (ItemStack s : drop)
+                    world.dropItemNaturally(location, s);
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /*
+     * @see com.nitnelave.CreeperHeal.block.Replaceable#update()
+     */
+    @Override
+    public void update()
+    {
+        blockState.getChunk().load();
```

Comment:
Can you comment on the chunk load? Why is it necessary? Why do you do it only there?

Reply:
I believe it's completely unnecessary, but that's [how it's been done](https://github.com/nitnelave/CreeperHeal/blob/master/src/main/java/com/nitnelave/CreeperHeal/block/CreeperBlock.java#L246).

Reply:
It's kinda hard to check, so let's keep it for now.

Reply:
Setting and getting blocks via the bukkit API always gets the chunk first, which always loads it if it's not loaded.

Reply:
Setting and getting blocks via the bukkit API always gets the chunk first, which always loads it if it's not loaded.

Reply:
(thats also a big part of why it's so slow for mass changes, that and relighting and physics)

Reply:
Then feel free to remove it!